app/assets/javascripts/lib/ranger_events_dispatcher.js

```diff
@@ -0,0 +1,42 @@
+var RangerWebSocket = function(url){
+    var conn = new WebSocket(url);
+
+    var callbacks = {};
+
+    this.bind = function(event_name, callback){
+        callbacks[event_name] = callbacks[event_name] || [];
+        callbacks[event_name].push(callback);
+        return this;// chainable
+    };
+
+    this.send = function(payload){
+        conn.send( payload ); // <= send JSON data to socket server
+        return this;
+    };
+
+    // dispatch to the right handlers
+    conn.onmessage = function(evt){
+        var json = JSON.parse(evt.data)
+        if (json instanceof Array) {
+            dispatch(json[0], json[1])
+        } else {
+            if (json['success']) {
+                console.log(json['success']['message'])
+            } else {
+                console.error(json['error']['message'])
```

Comment:
Can we let user know that he has problem with his JWT token? Like show maybe some dialog message about it.

Reply:
@kovetskiy I don't think showing dialog message is a good solution because user will always get a dialog whenever he refreshes the page, which I don't prefer. It's better if we display the errors on console OR we can show a notification showing that `there is a problem with JWT token` instead of dialog.

Reply:
Why will a user have a dialog every time when he refreshes the page? This json['error']['message'] will exist only if he was not able to authorize due to invalid JSON token.

Actually, a notification with error message would be enough too.

Reply:
Why will a user have a dialog every time when he refreshes the page? This json['error']['message'] will exist only if he was not able to authorize due to invalid JSON token.

Actually, a notification with error message would be enough too.

Reply:
Actually when we refresh the page it will establish a socket connection and try to authenticate. So, it will always show dialog with error message.

ok will show a notification with error message.